docs/sql/tutorials/slack-chatbot.mdx

```diff
@@ -0,0 +1,281 @@
+```
+
+Let's test this in action:
+
+```sql
+SELECT
+  messages, response
+FROM mindsdb.whizfizz_model
+WHERE messages = 'Hi, can you please explain me more about MindsDB?';
+```
+
+<p align="center">
+    <img src="/assets/SLBot-response2.png" />
+</p>
+
+You see the difference! Now, I'm getting excited, let's try again.
+
+```sql
+SELECT
+  messages, response
+FROM mindsdb.whizfizz_model
+WHERE messages = 'if a time-traveling astronaut had a dance-off with a black hole, what mind-bending moves would they showcase, and how would gravity groove to the rhythm?!';
+```
+<p align="center">
+    <img src="/assets/SLBot-response3.png" />
+</p>
+
+### 3. Let's Connect our GPT Model to Slack!
+
+The database has `channels` table. This table can be used to search for `channels`, `messages`, and `timestamps`, as well as to post `messages` into slack channels. These functionalities can also be done by using Slack API or Webhooks. 
+
+Let's query the user's question and see how our GPT model responds to it, by joining the model with the `channels` table:
+
+```sql
+SELECT
+    t.channel as channel,
+    t.messages as input_text, 
+    r.response as message
+FROM mindsdb_slack.channels as t
+JOIN mindsdb.whizfizz_model as r
+WHERE t.channel = '<channel-name>'
+LIMIT 3;
+```
+
+### 4. Posting Messages using SQL
+
+We want to respond to the user's questions by posting the output of our newly created **WhizFizz** Model. Let's post the message by querying and joining the user's questions to our model:
+
+```sql
+INSERT INTO mindsdb_slack.channels(channel, message)
+  SELECT
+    t.channel as channel,
+    t.messages as input_text, 
+    r.response as message
+  FROM mindsdb_slack.channels as t
+  JOIN mindsdb.whizfizz_model as r
+  WHERE t.channel = '<channel-name>'
+  LIMIT 3;
+```
+
+Works like a charm!!
+
+<p align="center">
+    <img src="/assets/SLBot-response4.png" />
+</p>
+
+### 5. Let's automate this
+
+We will `CREATE JOB` that means the query which we will give, will execute after some interval. Let's schedule a job every hour. The job will do the following:
+    1. Check for new messages
+    2. Generate an appropriate response
+    3. Post the response into the channel
+
+Let's do it in single SQL statement:
+
+```sql
+CREATE JOB mindsdb.gpt4_slack_job AS (
+   -- insert into channels the output of joining model and new responses
+  INSERT INTO mindsdb_slack.channels(channel, message)
+  SELECT
+    t.channel as channel,
+    t.messages as input_text, 
+    r.response as message
+  FROM mindsdb_slack.channels as t
+  JOIN mindsdb.whizfizz_model as r
+  WHERE t.channel = '<channel-name>'
```

Comment:
Every time this job is executed, the bot replies to the first ten messages.

It would be better to define in the WHERE clause to answer only the latest messages, like this:
```
WHERE t.message_created_at > "{{PREVIOUS_START_DATETIME}}"
```
Assuming there is a column (like `message_created_at`) that stores date/time when the message was sent.